powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/SignatureServiceBehavior.java

```diff
@@ -335,11 +336,18 @@ private ValidateSignatureResponse validateSignature(ActivationRecordEntity activ
         List<SecretKey> signatureKeys = powerAuthServerKeyFactory.keysForSignatureType(powerAuthSignatureTypes, masterSecretKey);
 
         // Get activation version and validate it
-        Integer version = activation.getVersion();
-        if (version == null || version < 2 || version > 3) {
+        Integer signatureVersion = activation.getVersion();
+        if (signatureVersion == null || signatureVersion < 2 || signatureVersion > 3) {
```

Comment:
This feels a bit too hardcoded to me... At least, isolate it in a separate method `boolean isValidSignatureVersion(version)`...